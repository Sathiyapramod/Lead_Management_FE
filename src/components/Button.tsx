import React from "react";
import { twMerge } from "tailwind-merge";

interface AppButton {
    theme: string;
    content: string;
    classname?: string;
    type?: "button" | "submit" | "reset";
    onClick?: (e: React.MouseEvent<HTMLButtonElement>) => Promise<void> | void;
    disabled?: boolean;
    src?: string;
}

function Button({ theme, content, onClick, classname, type, src, disabled }: AppButton) {
    return (
        <button
            onClick={onClick}
            type={type ?? "button"}
            className={twMerge(
                "lg:w-fit h-[48px] px-[16px] py-[5px] text-xs font-bold rounded-lg border border-black",
                theme === "dark" ? "bg-black text-white" : "bg-white text-black",
                classname
            )}
            disabled={disabled}
        >
            {content !== "" ? (
                content.toUpperCase()
            ) : (
                <img src={src} alt={"call_image"} width={35} height={35} />
            )}
        </button>
    );
}

export default Button;
