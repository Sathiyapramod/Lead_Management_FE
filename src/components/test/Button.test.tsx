import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import Button from "../Button";

describe("Button Component", () => {
    // test on button text content
    test("renders button with text content", () => {
        render(<Button theme="light" content="Click Me" />);
        const buttonElem = screen.getByRole("button", { name: /click me/i });
        expect(buttonElem).toBeInTheDocument();
        expect(buttonElem).toHaveTextContent("CLICK ME");
    });

    // test on disabled props passed
    test("button is disabled when disabled prop is true", () => {
        render(<Button theme="light" content="Click Me" disabled={true} />);

        const buttonElement = screen.getByRole("button", { name: /click me/i });
        expect(buttonElement).toBeDisabled();
    });

    test("button is disabled when disabled prop is false", () => {
        render(<Button theme="light" content="Click Me" disabled={false} />);

        const buttonElement = screen.getByRole("button", { name: /click me/i });
        expect(buttonElement).toBeEnabled();
    });
    // test on onClick event handler
    test("call onClick handler once the button is clicked", () => {
        const handleClick = jest.fn();
        render(<Button theme="light" content="Click Me" onClick={handleClick} />);

        const buttonElement = screen.getByRole("button", { name: /click me/i });
        fireEvent.click(buttonElement);

        expect(handleClick).toHaveBeenCalledTimes(1);
    });

    //test on classname passed as prop
    test("applies correct classes based on theme prop", () => {
        const { rerender } = render(<Button theme="dark" content="Dark Theme" />);

        const buttonElement = screen.getByRole("button", { name: /dark theme/i });
        expect(buttonElement).toHaveClass("bg-black text-white");

        rerender(<Button theme="light" content="Light Theme" />);
        expect(buttonElement).toHaveClass("bg-white text-black");
    });
});
