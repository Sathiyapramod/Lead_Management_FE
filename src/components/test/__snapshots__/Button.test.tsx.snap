// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component match snapshot 1`] = `[Function]`;
